## Banking Data - Descriptive Analysis


In [2]:
import pandas as pd

In [3]:
path_to_file = "banklist.csv" # YOU MIGHT NEED TO UPDATE THIS TO REFLECT THE PATH TO THE DATASET

In [4]:
df=pd.read_csv(path_to_file)
df.head()


Bank Name         City  ST   CERT  \
0                Washington Federal Bank for Savings      Chicago  IL  30570   
1    The Farmers and Merchants State Bank of Argonia      Argonia  KS  17719   
2                                Fayette County Bank   Saint Elmo  IL   1802   
3  Guaranty Bank, (d/b/a BestBank in Georgia & Mi...    Milwaukee  WI  30003   
4                                     First NBC Bank  New Orleans  LA  58302   

                 Acquiring Institution Closing Date Updated Date  
0                   Royal Savings Bank    15-Dec-17    21-Feb-18  
1                          Conway Bank    13-Oct-17    21-Feb-18  
2            United Fidelity Bank, fsb    26-May-17    26-Jul-17  
3  First-Citizens Bank & Trust Company     5-May-17    26-Jul-17  
4                         Whitney Bank    28-Apr-17     5-Dec-17

This data set includes US banks which have failed since October 1, 2000. <br>

- The `Bank Name` is the name of the failed bank. <br>
- The `Acquiring Institution` represents the bank that bought the failed bank. <br>
- The `CERT` represents the FDIC Certificate Number - a unique number assigned by the FDIC used to identify institutions and for the issuance of insurance certificates. <br>
- The `Closing Date` column stores the date that the acquisition closed. <br>
- The `Updated Date` stores the date at which the last update for the deal occurred. <br>
<br>

You can find more information about this dataset here: https://www.fdic.gov/bank/individual/failed/ <br>


#### How many rows/columns  are there in the dataset?


In [4]:
df.shape

(555, 7)

There are 555 rows and 7 columns.

#### Determine if the dataset has any missing values in the categorical columns. If there are, replace them with a "M" value. 

In [68]:
df.isna().sum()
# same: df.isnull().sum()

Bank Name                0
City                     0
ST                       0
CERT                     0
Acquiring Institution    0
Closing Date             0
Updated Date             0
dtype: int64

#### There is no missing value in this dataset.

#### What are the top 4 acquiring institutions? <br>
Note that the value `No Acquirer` means the failed bank was not acquired so this is not an "acquiring institution".

In [6]:
df['Acquiring Institution'].value_counts()[:5]


No Acquirer                            31
State Bank and Trust Company           12
First-Citizens Bank & Trust Company    11
Ameris Bank                            10
U.S. Bank N.A.                          9
Name: Acquiring Institution, dtype: int64

In [169]:
df['Acquiring Institution'].value_counts().index.tolist()[1:5]

['State Bank and Trust Company',
 'First-Citizens Bank & Trust Company',
 'Ameris Bank',
 'U.S. Bank N.A.']

#### So, the top 4 Acquiring Institution are:
State Bank and Trust Company           12,
First-Citizens Bank & Trust Company    11,
Ameris Bank                            10,
U.S. Bank N.A.                          9.

#### What are the 3 states that had the highest number of bank failures and how many bank failures did they have? <br>
#### What are the states with the lowest number of bank failures and how many bank failures did they have?<br>

In [30]:
df['ST'].value_counts().head(3)

GA    93
FL    75
IL    69
Name: ST, dtype: int64

In [22]:
df['ST'].value_counts().tail(10)

IA    2
ID    2
MS    2
KY    2
HI    1
NH    1
WV    1
WY    1
MA    1
SD    1
Name: ST, dtype: int64

The lowest number of bank failures is 1. These states are HI,   NH,  WV,     WY,    MA and SD.

#### How many distinct states are there in the dataset? <br>

In [34]:
df['ST'].nunique()

44

#### How many city names start with the letter 'A'?

In [115]:
# df[df['City'].str.startswith('A')]
df[df['City'].str.startswith('A')]['City'].nunique()

19

In [114]:
# Approach 2:
city_unique = pd.Series(df['City'].unique().tolist())
city_unique[city_unique.str.startswith('A')].count()

19

#### What are the cities whose names consist of 2 words and how many are there? (Note, take care of duplicates.)

In [164]:
pd.Series(df['City'].unique()).str.split()
index = pd.Series(df['City'].unique()).str.split().str.len() == 2
index

0      False
1      False
2       True
3      False
4       True
       ...  
421    False
422    False
423    False
424    False
425    False
Length: 426, dtype: bool

In [163]:
city_unique.loc[index]

2              Saint Elmo
4             New Orleans
5      Cottonwood Heights
12         Peachtree City
14               San Juan
              ...        
388           Culver City
396         Newport Beach
408          White Plains
417       Shelby Township
419         Sierra Blanca
Length: 108, dtype: object

#### There are 108 cities whose name contains 2 words. They are:

In [165]:
city_unique.loc[index].tolist()

['Saint Elmo',
 'New Orleans',
 'Cottonwood Heights',
 'Peachtree City',
 'San Juan',
 'Palm Desert',
 'Rising Sun',
 'Fort Lauderdale',
 'Bel Air',
 'El Reno',
 'West Chicago',
 'The Woodlands',
 'Fort Myers',
 'Orange Park',
 'Gold Canyon',
 'University Place',
 'Sunrise Beach',
 'St. Louis',
 'North Lauderdale',
 'Pawleys Island',
 'Maple Grove',
 'Fort Lee',
 'Rock Spring',
 'Little Falls',
 'Hoffman Estates',
 'Forest Lake',
 'Saint George',
 'Des Plaines',
 'Greenwood Village',
 'Nevada City',
 'Palm Beach',
 'Huntingdon Valley',
 'Apollo Beach',
 'Castle Rock',
 'Cocoa Beach',
 'Mount Clemens',
 'Winter Park',
 'East Ellijay',
 'Las Vegas',
 'Western Springs',
 'Wood Dale',
 'St. Charles',
 'Palm Springs',
 'Port Orange',
 'Lino Lakes',
 'Coral Gables',
 'Farmington Hills',
 'Bala Cynwyd',
 'Woodland Hills',
 'Overland Park',
 'Jefferson City',
 'West Allis',
 'Cherry Hill',
 'Palos Heights',
 'Cave Junction',
 'New Prague',
 'Sylvan Grove',
 'Port Chester',
 'Granite Bay',
 'Sa

In [16]:
df.ST.value_counts()

GA    93
FL    75
IL    69
CA    41
MN    23
WA    19
MO    16
AZ    16
MI    14
NV    12
TX    12
WI    11
CO    10
MD    10
SC    10
PA    10
KS    10
OH     8
UT     8
AL     7
NC     7
OK     7
NJ     7
TN     7
OR     6
NY     5
VA     5
LA     4
AR     4
PR     4
NM     3
IN     3
NE     3
IA     2
CT     2
MS     2
ID     2
KY     2
MA     1
WV     1
NH     1
SD     1
HI     1
WY     1
Name: ST, dtype: int64

In [22]:
df.City.value_counts().head(10)

Chicago        20
Atlanta        10
Phoenix         6
Naples          5
Las Vegas       4
Alpharetta      4
Los Angeles     4
Miami           4
Bradenton       4
Scottsdale      4
Name: City, dtype: int64

In [18]:
df['Closing Date'].min(),df['Closing Date'].max()

('1-Aug-08', '9-Sep-11')